### PR TITLE
Feature/create header

### PR DESCRIPTION
**Pull request by DenysBabenko29**

### CODE REVIEW

File: src/components/Footer/Footer.tsx [hunk 1]
```diff
@@ -179,7 +179,7 @@ const Footer: React.FC = (): JSX.Element => {
 							</ul>
 						</div>
 						<div className="footer__widget">
-							<h2 className="footer__widget--title text-ofwhite h3">
+							<h2 className="footer__widget--title text-ofwhite h3 cla">
```
Comment (Igor-Klein-1988) [inline src/components/Footer/Footer.tsx]: What is `cla`?

File: src/components/Header/set_language/SetLanguage.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+			<ul className="d-flex align-items-center">
+				<li className="language__currency--list">
+					<img className="language__switcher--icon__img" src={language_icon} alt="language"></img>
+					<span
+						className="language_text_set"
+						onClick={() => {
+							seetIsActive(!isActive);
+						}}
+					>
+						{t('language')}
```
Comment (Igor-Klein-1988) [inline src/components/Header/set_language/SetLanguage.tsx]: Could you use value from `locales`?

File: public/locales/de.json [hunk 3]
```diff
@@ -14,11 +14,27 @@
...
+	},
+	"header" : {
+		"welcome": "Willkommen im Netmark Online-Shop",
+		"language": "Deutsch",
+		"login": "Anmeldung",
+		"register": "Registrieren",
+		"contact_us": "Kontakt",
+		"about": "Über Uns",
+		"faq": "FAQ",
+		"auction": "Auction",
```
Comment (Igor-Klein-1988) [inline public/locales/de.json]: Auktion

File: src/components/Footer/Footer.tsx (hunk 1 above)
Comment (DenysBabenko29) [reply src/components/Footer/Footer.tsx]: test class, deleted

File: public/locales/ru.json [hunk 4]
```diff
@@ -16,15 +16,41 @@
...
+		"contact_us": "Контакты",
 		"portfolio": "Портфолио",
 		"privacy_policy": "Политика Конфиденциальности",
 		"copyright": "Авторские права © 2023",
 		"title": "Allcom",
 		"reserved": ". Все права защищены."
 	},
+	"header" : {
+		"welcome": "Добро пожаловать в интернет-магазин Netmark",
+		"language": "Русский",
```
Comment (Igor-Klein-1988) [inline public/locales/ru.json]: useless

File: public/locales/de.json [hunk 5]
```diff
@@ -18,13 +18,39 @@
...
+		"contact_us": "Kontakt",
 		"portfolio": "Portfolio",
 		"privacy_policy": "Datenschutzrichtlinie",
 		"copyright": "Urheberrechte © 2023",
 		"title": "Allcom",
 		"reserved": ". Alle Rechte vorbehalten."
 	},
+	"header" : {
+		"welcome": "Willkommen im Netmark Online-Shop",
+		"language": "Deutsch",
```
Comment (Igor-Klein-1988) [inline public/locales/de.json]: useless

File: public/locales/de.json [hunk 6]
```diff
@@ -18,13 +18,39 @@
...
+		"login": "Anmeldung",
+		"register": "Registrieren",
+		"contact_us": "Kontakt",
+		"about": "Über Uns",
+		"faq": "FAQ",
+		"auction": "Auktion",
+		"categories": "Kategorien",
+		"search": "Suchen",
+		"cart": "Cart",
```
Comment (Igor-Klein-1988) [inline public/locales/de.json]: Warenkorb

File: public/locales/de.json [hunk 7]
```diff
@@ -18,13 +18,39 @@
...
 		"portfolio": "Portfolio",
 		"privacy_policy": "Datenschutzrichtlinie",
 		"copyright": "Urheberrechte © 2023",
 		"title": "Allcom",
 		"reserved": ". Alle Rechte vorbehalten."
 	},
+	"header" : {
+		"welcome": "Willkommen im Netmark Online-Shop",
+		"language": "Deutsch",
+		"login": "Anmeldung",
```
Comment (Igor-Klein-1988) [inline public/locales/de.json]: Einloggen

File: public/locales/de.json [hunk 8]
```diff
@@ -18,13 +18,39 @@
...
+		"register": "Registrieren",
+		"contact_us": "Kontakt",
+		"about": "Über Uns",
+		"faq": "FAQ",
+		"auction": "Auktion",
+		"categories": "Kategorien",
+		"search": "Suchen",
+		"cart": "Cart",
+		"shopping_cart": "Einkaufswagen",
```
Comment (Igor-Klein-1988) [inline public/locales/de.json]: use only "cart": "Warenkorb".

File: public/locales/en.json [hunk 9]
```diff
@@ -25,6 +25,31 @@
...
+		"welcome": "Welcome to Netmark online Store",
+		"language": "Englisch",
+		"about": "About Us",
+		"faq": "FAQ",
+		"contact_us": "Contact",
+		"register": "Register",
+		"auction": "Auction",
+		"categories": "Categories",
+		"search": "Search",
+		"shopping_cart": "Shopping Cart",
```
Comment (Igor-Klein-1988) [inline public/locales/en.json]: forgot "cart"

File: public/locales/ru.json [hunk 10]
```diff
@@ -16,15 +16,41 @@
 		"shopping_cart": "Корзина",
```
Comment (Igor-Klein-1988) [inline public/locales/ru.json]: use "cart"

File: public/locales/ru.json [hunk 11]
```diff
@@ -16,15 +16,41 @@
...
 		"privacy_policy": "Политика Конфиденциальности",
 		"copyright": "Авторские права © 2023",
 		"title": "Allcom",
 		"reserved": ". Все права защищены."
 	},
+	"header" : {
+		"welcome": "Добро пожаловать в интернет-магазин Netmark",
+		"language": "Русский",
+		"about": "О Нас",
+		"login": "Авторизоваться",
```
Comment (Igor-Klein-1988) [inline public/locales/ru.json]: Войти

File: public/locales/ru.json [hunk 12]
```diff
@@ -16,15 +16,41 @@
...
+		"register": "Регистрация",
+		"contact_us": "Контакты",
+		"faq": "Вопросы",
+		"auction": "Аукцион",
+		"categories": "Категории",
+		"search": "Поиск",
+		"shopping_cart": "Корзина",
```
Comment (Igor-Klein-1988) [inline public/locales/ru.json]: "cart"

File: src/components/Header/Header.module.scss [hunk 13]
```diff
@@ -0,0 +1,1389 @@
...
+    2. Header css
+*/
+
+.img_logo {
+  width: 120px;
+  border-radius: 10px;
+  cursor: pointer;
+}
+
+.offcanvas__header--menu__open {
```
Comment (Igor-Klein-1988) [inline src/components/Header/Header.module.scss]: make name `header__offcanvas--menu__open `

File: src/components/Header/Header.module.scss [hunk 14]
```diff
@@ -0,0 +1,1389 @@
...
+  display: none;
+  @media only screen and (max-width: 991px) {
+    display: block;
+  }
+  &--svg {
+    width: 32px;
+  }
+}
+
+.offcanvas__header {
```
Comment (Igor-Klein-1988) [inline src/components/Header/Header.module.scss]: same

File: src/components/Header/Header.module.scss [hunk 15]
```diff
@@ -0,0 +1,1389 @@
+/*
+    2. Header css
+*/
+
+.img_logo {
```
Comment (Igor-Klein-1988) [inline src/components/Header/Header.module.scss]: make name `.header__img_logo {`

File: src/components/Header/Header.tsx [hunk 16]
```diff
@@ -0,0 +1,840 @@
+import { useTranslation } from 'react-i18next';
+import { useState, useEffect } from 'react';
+import './Header.module.scss';
```
Comment (Igor-Klein-1988) [inline src/components/Header/Header.tsx]: empty line

File: src/components/Header/set_language/SetLanguage.tsx [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+		en: { title: 'English' },
+		de: { title: 'Deutsch' },
+		ru: { title: 'Русский' },
+	};
+	const { i18n } = useTranslation();
+
+	const [isActive, seetIsActive] = useState(false);
+
+	return (
+		<div className={`language__currency ${isOpen.isOpen} d-lg-block`} style={{ marginRight: '3%' }}>
```
Comment (Igor-Klein-1988) [inline src/components/Header/set_language/SetLanguage.tsx]: style={{ marginRight: '3%' }} move to *.scss

File: src/components/Header/set_language/SetLanguage.tsx [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+					>
+						{locales[i18n.language as keyof typeof locales].title}
+					</span>
+					<div className={`dropdown__language ${isActive ? 'active' : ''}`}>
+						<ul>
+							{Object.keys(locales).map((locale) => (
+								<li
+									key={locale}
+									className="language__text language_trans"
+									style={{
```
Comment (Igor-Klein-1988) [inline src/components/Header/set_language/SetLanguage.tsx:38]: same

File: public/locales/de.json [hunk 19]
```diff
@@ -18,13 +18,39 @@
 		"register": "Registrieren",
 		"wishlist": "Wunschzettel",
```
Comment (Igor-Klein-1988) [inline public/locales/de.json]: "Merkzettel"

File: public/locales/ru.json (hunk 4 above)
Comment (DenysBabenko29) [reply public/locales/ru.json:31]: language => useles?

File: public/locales/de.json (hunk 19 above)
Comment (DenysBabenko29) [reply public/locales/de.json]: fixed

File: src/components/Header/set_language/SetLanguage.tsx (hunk 17 above)
Comment (DenysBabenko29) [reply src/components/Header/set_language/SetLanguage.tsx]: fixed

File: src/components/Header/Header.tsx (hunk 16 above)
Comment (DenysBabenko29) [reply src/components/Header/Header.tsx]: fixed

File: src/components/Header/Header.module.scss (hunk 15 above)
Comment (DenysBabenko29) [reply src/components/Header/Header.module.scss]: fixed

File: public/locales/de.json (hunk 8 above)
Comment (DenysBabenko29) [reply public/locales/de.json]: fixed

File: public/locales/de.json (hunk 7 above)
Comment (DenysBabenko29) [reply public/locales/de.json]: fixed

File: public/locales/de.json (hunk 6 above)
Comment (DenysBabenko29) [reply public/locales/de.json]: fixed

File: public/locales/de.json (hunk 5 above)
Comment (DenysBabenko29) [reply public/locales/de.json]: fixed

File: public/locales/ru.json (hunk 4 above)
Comment (DenysBabenko29) [reply public/locales/ru.json]: fixed

File: src/components/Header/Header.module.scss [hunk 20]
```diff
@@ -0,0 +1,1394 @@
+/*
+    2. Header css
+*/
+
+.language_style {
```
Comment (Igor-Klein-1988) [inline src/components/Header/Header.module.scss]: header__language_style

File: src/components/Header/Header.tsx [hunk 21]
```diff
@@ -0,0 +1,839 @@
...
+											src="assets/img/logo/nav-log.png"
+											alt="logo"
+										></img>
+									</a>
+								</h1>
+							</div>
+							<div className="header__search--widget header__sticky--none d-none d-lg-block mb-15">
+								<form className="d-flex header__search--form" action="#">
+									<div className="header__select--categories select">
+										<select className="header__select--inner">
```
Comment (Igor-Klein-1988) [inline src/components/Header/Header.tsx]: use react-select

File: src/assets/scss/elements/_footer.module.scss [hunk 22]
```diff
@@ -25,16 +26,16 @@
...
   @media only screen and (min-width: 992px) and (max-width: 1199px) {
-    width: 31%;
+    width: 35%;
   }
   @media only screen and (min-width: 768px) and (max-width: 991px) {
-    width: 47%;
+    width: 37%;
   }
 }
 .social__title {
```
Comment (Igor-Klein-1988) [inline src/assets/scss/elements/_footer.module.scss]: footer__social__title

File: src/assets/scss/elements/_footer.module.scss [hunk 23]
```diff
@@ -0,0 +1,400 @@
+.footer__section {
+	@media only screen and (max-width: 991px) {
+		padding-bottom: 7rem;
+	}
+}
+
+.main__footer {
```
Comment (Igor-Klein-1988) [inline src/assets/scss/elements/_footer.module.scss]: .footer__main

File: src/assets/scss/elements/_footer.module.scss [hunk 24]
```diff
@@ -0,0 +1,400 @@
...
+	}
+	@media only screen and (min-width: 768px) and (max-width: 991px) {
+		width: 37%;
+	}
+}
+.footer_social__title {
+	font-size: 1.8rem;
+	font-weight: 400;
+}
+.social__shear {
```
Comment (Igor-Klein-1988) [inline src/assets/scss/elements/_footer.module.scss]: footer_social__shear

File: src/assets/scss/elements/_footer.module.scss [hunk 25]
```diff
@@ -0,0 +1,400 @@
...
+		visibility: visible;
+	}
+}
+
+.footer__widget--inner {
+	@media only screen and (max-width: 767px) {
+		display: none;
+	}
+}
+.newsletter__subscribe {
```
Comment (Igor-Klein-1988) [inline src/assets/scss/elements/_footer.module.scss]: same

File: src/assets/scss/elements/_footer.module.scss [hunk 26]
```diff
@@ -0,0 +1,400 @@
...
+}
+
+.footer__instagram--list {
+	margin-bottom: 1rem;
+	&:last-child {
+		margin-bottom: 0;
+	}
+}
+
+.instagram__thumbnail {
```
Comment (Igor-Klein-1988) [inline src/assets/scss/elements/_footer.module.scss]: same
do you use it?

File: src/assets/scss/elements/_footer.module.scss [hunk 27]
```diff
@@ -0,0 +1,400 @@
...
+	& a {
+		display: block;
+		& img {
+			display: block;
+			width: 8rem;
+		}
+	}
+}
+
+.copyright__content {
```
Comment (Igor-Klein-1988) [inline src/assets/scss/elements/_footer.module.scss]: same

File: src/assets/scss/elements/_footer.module.scss [hunk 28]
```diff
@@ -0,0 +1,400 @@
...
+		justify-content: center;
+		margin-bottom: 2rem;
+	}
+	& ul {
+		@media only screen and (max-width: 1199px) {
+			justify-content: center;
+		}
+	}
+}
+.copyright__content.style3 {
```
Comment (Igor-Klein-1988) [inline src/assets/scss/elements/_footer.module.scss]: same

**Review by Igor-Klein-1988 — APPROVED** [review]

**Pull request merged**